seqr/views/apis/saved_variant_api.py

```diff
@@ -92,14 +93,39 @@ def create_saved_variant_handler(request):
 @login_required(login_url=API_LOGIN_REQUIRED_URL)
 @csrf_exempt
 def create_variant_note_handler(request, variant_guid):
+    save_as_gene_note = json.loads(request.body).get('saveAsGeneNote')
     saved_variant = SavedVariant.objects.get(guid=variant_guid)
     check_permissions(saved_variant.project, request.user, CAN_VIEW)
 
+    if save_as_gene_note:
+        gene_id = json.loads(saved_variant.saved_variant_json)['mainTranscript']['geneId']
+        _create_gene_note(gene_id, json.loads(request.body), request.user)
+
+    gene_note = {gene_id: {
+        'notes': _get_gene_notes(gene_id, request.user)}} if save_as_gene_note else {}
+
     _create_variant_note(saved_variant, json.loads(request.body), request.user)
+    variant_note = {variant_guid: {
+        'notes': [get_json_for_variant_note(tag) for tag in saved_variant.variantnote_set.all()]},
+    }
 
-    return create_json_response({'savedVariantsByGuid': {variant_guid: {
-        'notes': [get_json_for_variant_note(tag) for tag in saved_variant.variantnote_set.all()]
-    }}})
+    return create_json_response({
+        'savedVariantsByGuid': variant_note,
+        'genesById': gene_note
+    })
+
+
+def _create_gene_note(gene_id, note_json, user):
```

Comment:
I don't think making these into helper functions adds much. If there was a bunch of things being done in the function it would add readability, but these are both one liners. If there is shared functionality and the helper function is used in multiple places, thats good because it makes it easier to make updates in the future. But in general, a one line helper function thats only called once is less readable than just running the code in line